doc/source/design/design-timestep.rst

```diff
@@ -0,0 +1,202 @@
+Super-cycling Gravity
+=====================
+
+As a driving use-case, we wish to super-cycle gravity solves with
+respect to hydrodynamics. The rationale is that gravity solves are
+relatively costly, and gravitational potentials typically evolve at
+longer time-scales than hydro fields.
+
+To super-cycle gravity, the user supplies a single integer parameter
+:code:`Method : gravity : max_supercycle` that defines how frequently
+the :code:`"gravity"` method is to be called with respect to other
+methods. The default value of :code:`1` effectively turns off
+super-cycling, whereas a value of :math:`k` allows other methods to be
+called *up to* :math:`k` times for each call to the gravity
+solve. Note that the maximum local gravity time-step restriction on
+each block will still always be satisfied, even if the
+:code:`max_supercycle` parameter is set to an arbitrarily large value.
+
+For cycles where the gravitational potential solve is not performed,
+extrapolations of saved fields are used to approximate the current
+fields.  Which fields to be extrapolated is determined by the
+:code:`Method : gravity : type_super` parameter, which can be either
+:code:`"potential"` or :code:`"accelerations"`. The default is
+:code:`"accelerations"`, which is preferred since extrapolating
+potentials can introduce grid effects at refinement level
+boundaries. The figure below illustrates these two variations of
+super-cycling gravity as compared to non super-cycled gravity.
+
+.. figure:: supercycle-gravity.png
+           :width: 1375
+
+           This figure illustrates standard versus super-cycled
+           gravity. Time runs horizontally left-to-right. Green
+           circles represent linear solves for gravitational
+           potential, magenta squares represent hydro solves, and blue
+           triangles represent accelerations computed from potentials.
+           White fields are extrapolated from previously computed
+           values.  The left figure shows standard gravity with
+           gravity solves every cycle; the figures to the right show
+           super-cycled gravity, with the upper-right variation
+           showing extrapolated gravitational potentials, and the
+           lower-right variation showing extrapolated accelerations.
+
+
+------------------------
+non super-cycled gravity
+------------------------
+
+The gravitational potential is used to compute gravitational
+accelerations, which are applied to all gravitating particles and
+fields. We use a leapfrog integration scheme because it is 2nd order
+accurate (small short-scale errors) and symplectic (qualitatively
+correct long-scale behavior). This involves computing time-centered
+accelerations. Since we start with non-time centered values, we must
+either extrapolate the input (density) to the gravity solve, or
+extrapolate the output (potential) from the gravity solve. In both
+ENZO and Enzo-E, the extrapolation is done before the gravity solve.
```

Comment:
Extrapolating the density involves the `Method : pm_deposit : alpha` parameter, right?

If so then our approach isn't robust in the general case. When I implemented self-gravity source terms for VL+CT, there was a whole saga where I was getting bizare looking results for a standard test problem (i.e. the "stable linear Jeans wave" that is inclined with respect to the grid and for which we can perform an analytic convergence study). Ultimately, the solution was to stop extrapolating. I suspect that we would find similar issues with PPM if we ran a similar test

In that scenario, there were no particles in that case. It's plausible that if particles dominate the short-range gravitational potential that extrapolation might be more robust.

Anyway, we probably don't want to go into detail here, but maybe we add something like the following at the end of this subsection?

```rst
.. caution:

   Some care is needed when applying extrapolation methods. For certain problems, they may not produce meaningful results.
```